File: Java/src/PowersOf2.java
import java.util.Scanner;

public class PowersOf2 {
    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter a number to check for power of 2: ");
        int userNum = in.nextInt();
        in.nextLine();
        int currPowerOfTwo = 1;
        while (currPowerOfTwo < userNum) {
            currPowerOfTwo *= 2;
        }
        if (userNum == currPowerOfTwo) {
            System.out.println("Your number is a power of 2");
        } else {
            System.out.println("Your number is not a power of 2");
        }

        in.close();
    }
}

File: Python/powers_of_2.py
userNum = int(input("Enter a number to check if it's a power of 2: "))
currPower = 1
while currPower < userNum:
    currPower *= 2

if currPower == userNum:
    print("Your number is a power of 2")
else:
    print("Your number is not a power of 2")
